### PR TITLE
Kickstarter Scraper

### PR DESCRIPTION
**Pull request by akashrajkn**
Contains only the scraper. I'll make a separate PR for the ip spoofing

### CODE REVIEW

File: forward43/kickstarter_scraper.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    with open(filepath, 'w+') as f:
+        json.dump(data, f)
+
+
+def generate_url(category, page):
+    '''
+    Generates a URL to scrape from. It's necessary that category and page parameters are not None
+    '''
+    random_seed   = 'seed='   + str(random.randint(1, MAX_VALUE))
+    random_woe_id = 'woe_id=' + str(random.randint(1, MAX_VALUE))
```
Comment (andrewsutjahjo) [inline forward43/kickstarter_scraper.py:36]: Interesting! Never heard of woe_id's before. 
A quick google showed that there's a lot more than that MAX_VALUE though: new york is `2459115` which is above the `MAX_VALUE`. And entering pseudorandom woe_id's myself gives me 0 results quite often, which might not be great given that we're dealing with rate-limiters. 

Maybe a way forward is to have a list of countries woe_id's and picking randomly from there, or setting the woe_id to 1 (everywhere, I believe?) and letting the random_seed do all the work of getting new projects

**Review by andrewsutjahjo — CHANGES_REQUESTED** [review]
One main comment on json/dict value getting,
and a question on the woe_id's.

Also a super minor comment: `f"{some_var}"` is faster then `"{}".format(some_var)`

File: forward43/kickstarter_scraper.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import requests
+from pathlib import Path
+
+from datetime       import datetime
+from urllib.request import urlopen
+
+
+MAX_VALUE                  = 65536
+KICKSTARTER_URL            = 'https://www.kickstarter.com/discover/advanced.json?'
+KICKSTARTER_DEFAULT_PARAMS = ['sort=popularity']
```
Comment (andrewsutjahjo) [inline forward43/kickstarter_scraper.py:15]: Oh, I'm not too sure about this popularity sorting: not sure how their popularity ranking and the random_seed interacts with each other. 
If popularity overpowers the random_seed, we could potentially be getting a lot of projects that overlap between runs, and missing lots of projects as we're only looking at the top 60 popular in that seed.

Comment (akashrajkn) [reply forward43/kickstarter_scraper.py:15]: Good point. I just checked that it's not a required parameter. I will get rid of it.
(An alternative would be to use `sort=newest`)

File: forward43/kickstarter_scraper.py (hunk 1 above)
Comment (akashrajkn) [reply forward43/kickstarter_scraper.py:36]: For now, I'll set woe_id to 1. 

Getting a random woe_id should be possible for later.

File: forward43/kickstarter_scraper.py [hunk 3]
```diff
@@ -52,22 +56,18 @@ def fetch_data(url):
...
-                'state'       : data['projects'][i]['state'],
-                'backers'     : str(data['projects'][i]['backers_count']),
-                'launch_date' : time.strftime('%c', time.localtime(data['projects'][i]['launched_at'])),
-                'deadline'    : time.strftime('%c', time.localtime(data['projects'][i]['deadline'])),
-                'url'         : data['projects'][i]['urls']['web']['project'],
-            }
+            details = { 'rank' : str(i + 1) }
+
+            for att in ATTRIBUTES:
+                details[att] = data['projects'][i].get(att, 'n.a.')
```
Comment (andrewsutjahjo) [inline forward43/kickstarter_scraper.py:62]: This'll overwrite the rank attribute for all details as `'n.a.'` , I'd say either remove rank from the `ATTRIBUTES` , or if you want it in the `ATTRIBUTES`, add the rank after this for loop :)

**Review by andrewsutjahjo — APPROVED** [review]
Small change needed for the rank, but after that it's good to go

File: forward43/kickstarter_scraper.py (hunk 3 above)
Comment (akashrajkn) [reply forward43/kickstarter_scraper.py:62]: Good catch! I removed it from ATTRIBUTES.

**Pull request merged**